plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/explorer/BaseExplorerContentProvider.java

```diff
@@ -182,121 +171,52 @@ protected IStatus run(IProgressMonitor monitor) {
 		job.schedule();
 		return stub;
 	}
-	
-	private void asyncViewerRefresh(final Runnable runner) {
-		Control control = viewer.getControl();
-		if(control.isDisposed()) {
-			return;
-		}
-		control.getDisplay().asyncExec(runner);
-	}
-	
-	protected void addChildrenToViewer(final Object parent, final Object ... objects) {
-		asyncViewerRefresh(new Runnable() {
-			@Override
-			public void run() {
-				synchronized (viewer) {
-					viewer.add(parent,objects);
-				}
-			}
-		});
-	}
-
-	protected void removeChildrenFromViewer(final Object parent, final Object ... objects) {
-		asyncViewerRefresh(new Runnable() {
-			@Override
-			public void run() {
-				synchronized (viewer) {
-					viewer.remove(parent, objects);
-				}
-			}
-		});
-	}
-	protected void updateChildrenFromViewer(final Object ... objects) {
-		asyncViewerRefresh(new Runnable() {
-			@Override
-			public void run() {
-				synchronized (viewer) {
-					viewer.update(objects, null);
-				}
-			}
-		});
-	}
 
 	protected void refreshViewer(final Object object) {
-		asyncViewerRefresh(new Runnable() {
-			@Override
-			public void run() {
-				synchronized (viewer) {
-					if(object != null) {
-						viewer.refresh(object);
-					}else {
-						viewer.refresh();
+		Control control = viewer.getControl();
+		if (!control.isDisposed()) {
+			control.getDisplay().asyncExec(new Runnable() {
+				@Override
+				public void run() {
+					if (!viewer.getControl().isDisposed()) {
+						if (object != null) {
+							viewer.refresh(object);
+						} else {
+							viewer.refresh();
+						}
 					}
 				}
-			}
-		});
-	}
-	
-	protected void expand(final Object element, final int level) {
-		viewer.getControl().getDisplay().syncExec(new Runnable() {
-			@Override
-			public void run() {
-				viewer.expandToLevel(element, level);
-			}
-		});
+			});
+		}
 	}
-	
+
 	@Override
 	public Object getParent(Object element) {
 		return null;
 	}
 
-	@Override
-	public void dispose() {
-		if(input != null) {
-			input.removeListener(connectionListener);
-		}
-	}
-	
-	private class ConnectionsRegistryListener implements IConnectionsRegistryListener{
```

Comment:
yes, please explain, I miss this insight, too, would love to understand this design choice.